README.md

```diff
@@ -128,6 +128,14 @@ A wide range of lockfiles are supported by utilizing this [lockfile package](htt
 $ osv-scanner --lockfile=/path/to/your/package-lock.json -L /path/to/another/Cargo.lock
 ```
 
+The scanner will attempt to automatically determine the parser to use for each
+file based on the filename - you can manually specify the parser to use for all
+files with the `-parse-as` flag:
+
+```bash
+$ osv-scanner --parse-as 'requirements.txt' --lockfile=/path/to/your/extra-requirements.txt
```

Comment:
The semantics of this (applies to every single file you scan) seems a bit limiting. 

Some alternatives that make this more flexible: 

## 1. Explicit argument for every lockfile type

```
osv-scanner --requirements-txt=/path/to/requirements.txt
osv-scanner --cargo-lock=/path/to/...
```

Downsides: This results in multiple flags that do the same thing (scanning a lockfile), which doesn't seem ideal. The format of the flag itself is also not obvious (i.e. "Cargo.lock" corresponds to "--cargo-lock"). 

## 2. A delimiter in the existing `lockfile` value.
```
osv-scanner --lockfile=PARSE-AS:requirements.txt:/path/to/....
````

Downsides: The string format is hard to read and easy to get wrong. 

## 3. Allow --parse-as to be overridable 

`--parse-as` would apply to all subsequent `--lockfile` arguments, and can be specified multiple times. 

```
osv-scanner --parse-as=requirements.txt --lockfile=1/2.txt  --lockfile=3/4.txt --parse-as=Cargo.lock --lockfile=5/Cargo.lock
```

The first two lockfiles (`1/2.txt`, `3/4.txt`) are parsed using `requirements.txt`. `5/Cargo.lock` is parsed using `Cargo.lock`  because we passed a subsequent `--parse-as` that overrides this. 

Option 3. Seems like the best one to me, in a way that's consistent with our existing CLI format, without introducing redundancies. WDYT?